pkg/nodeconfig/nodeconfig.go

```diff
@@ -184,6 +186,17 @@ func (nc *nodeConfig) Configure(ctx context.Context) error {
 			}
 		}
 
+		// Ensure node-specific RBAC exists
+		if err := nc.ensureNodeSpecificRBAC(ctx); err != nil {
```

Comment:
During the creation and deletion of RBAC resources (per-node) there is a change to race conditions if not handled atomically. A node may be compromised before RBAC is updated and retain broader permissions. Is it worth adding a test for this case?